### PR TITLE
Use user kubeconfig instead of admin kubeconfig

### PR DESCRIPTION
**Pull request by austintackaberry**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:
Uses user kubeconfig instead of admin kubeconfig to access the target cluster. It is needed because the admin kubeconfig contains a 2year cert that cannot be revoked.

Related to https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/2129

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

One issue with this PR is that CAPI controller logs into the API server of the cluster to get details of the nodes ([reconcileNodeRefs](https://github.com/kubernetes-sigs/cluster-api/blob/af33e43d891e3af3787e86bceabce6b8e052c984/exp/controllers/machinepool_controller_noderef.go#L48)). The change to user kubeconfig breaks this flow.

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note

```

### CODE REVIEW

File: azure/services/managedclusters/client.go [hunk 1]
```diff
@@ -65,7 +65,7 @@ func (ac *AzureClient) GetCredentials(ctx context.Context, resourceGroupName, na
 	ctx, _, done := tele.StartSpanWithLogger(ctx, "managedclusters.AzureClient.GetCredentials")
 	defer done()
 
-	credentialList, err := ac.managedclusters.ListClusterAdminCredentials(ctx, resourceGroupName, name, "")
+	credentialList, err := ac.managedclusters.ListClusterUserCredentials(ctx, resourceGroupName, name, "")
```
Comment (CecileRobertMichon) [inline azure/services/managedclusters/client.go:68]: func comment needs to be updated tto reflect this change

**Comment by CecileRobertMichon** [comment]
@austintackaberry thanks for the PR! Please sign the CLA so we can move forward

File: azure/services/managedclusters/client.go (hunk 1 above)
Comment (austintackaberry) [reply azure/services/managedclusters/client.go:68]: updated, thanks!

**Comment by austintackaberry** [comment]
@CecileRobertMichon I should now be in a group that has signed the CLA. How can I re-trigger the bot to check?

**Comment by CecileRobertMichon** [comment]
@austintackaberry can you please follow the instructions in https://github.com/kubernetes/community/blob/master/CLA.md? Commenting "/easycla" should retrigger the check, in theory, but not seeing it go green.

**Comment by CecileRobertMichon** [comment]
also please squash commits

/ok-to-test

**Comment by CecileRobertMichon** [comment]
@alexeldeib thoughts on this one?

**Comment by CecileRobertMichon** [comment]
@alexeldeib thoughts on this one?

**Comment by alexeldeib** [comment]
how does this work for a non-AAD enabled cluster? I admittedly am not super familiar with the func you switched to, but I thought it fetched credentials for the AAD user and requires RBAC role for access?

**Comment by austintackaberry** [comment]
> how does this work for a non-AAD enabled cluster?
Yeah, I don't think using user kubeconfig would work for non-aad clusters

> I am wondering what rights the e2e SP has over the subscription and if it's simply a consequence of that providing sufficient access, which I don't think we can rely on?
As long as the SP is included in the `AADProfile.AdminGroupObjectIDs`, then it will have admin access to all target clusters

**Comment by alexeldeib** [comment]
I think this change probably needs to account for differentiating AAD and non AAD clusters. Non AAD clusters will still require the admin kubeconfig.

could you also doc the requirement for the management cluster SP to use user credentials if the workload cluster is aad-enabled? That seems like a natural foot gun. @CecileRobertMichon curious on your thoughts about the UX here.

**Comment by jackfrancis** [comment]
@austintackaberry I think we can probably introduce this functionality with the addition of `DisableUserAccounts` as a first-class capz + AKS configuration. I've begun work on this here:

https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/2512

**Comment by CecileRobertMichon** [comment]
/close

Doing some PR queue cleanup 🧹, please feel free to reopen when this is ready for review

**Pull request closed without merging**